lib/ansible/executor/task_executor.py

```diff
@@ -476,16 +477,16 @@ def _execute(self, variables=None):
                 self._play_context.remote_addr != self._connection._play_context.remote_addr):
             self._connection = self._get_connection(variables=variables, templar=templar)
             hostvars = variables.get('hostvars', None)
-            if hostvars:
+            # only template the vars if the connection actually implements set_host_overrides
+            # NB: this is expensive, and should be removed once connection-specific vars are being handled by play_context
+            if hostvars and getattr(self._connection, 'set_host_overrides') != getattr(ConnectionBase, 'set_host_overrides'):
```

Comment:
Just an artifact from a previous iteration (though the AttributeError would happen either way if it were missing). We should never hit the missing case, since it has to be ConnectionBase (it'll just return the impl from the base instead if it's not overridden), but just in case, I changed the impl to default None if the leaf plugin class somehow doesn't have the method defined.